src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -626,6 +729,33 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     void TermControl::_KeyDownHandler(winrt::Windows::Foundation::IInspectable const& /*sender*/,
                                       Input::KeyRoutedEventArgs const& e)
     {
+        // If Esc is pressed and the search box is opened, we close the search box
+        if (_searchBox && e.OriginalKey() == winrt::Windows::System::VirtualKey::Escape)
+        {
+            _CloseSearchBoxControl(nullptr, nullptr);
+            return;
+        }
+
+        // If the current focused element is a kind of input box, or it is an
+        // child element of searchbox, we do not send this event up to
+        // terminal
+        auto focusedElement = Input::FocusManager::GetFocusedElement(_root.XamlRoot());
+        bool isWinthinSearchBox = false;
+        if (_searchBox)
+        {
+            isWinthinSearchBox = _searchBox.CheckSearchBoxElement(focusedElement);

```

Comment:
I wonder if XAML UserControl has this method ? I only found it in Forms control.

Reply:
This will still need to be a function you implement (and can work the same as what you currently have).  I'm just proposing a name change and letting it get the focus element internally, so the caller has this simpler API.

Reply:
changed